### ISSUE TITLE
QST:"Shapely only supports coordinates in X,Y order" - why does it imply lon/lat axis order?

### ISSUE THREAD

**Issue by idanmiara** [shapely/shapely#1751]
Questions about the following statement from the manual:
https://github.com/shapely/shapely/blob/main/docs/manual.rst
`It is important to note that in the example above, the always_xy kwarg is required as Shapely only supports coordinates in X,Y order, and in PROJ 6 the WGS84 CRS uses the EPSG-defined Lat/Lon coordinate order instead of the expected Lon/Lat.`

I'm developing a software internally stores coordinates in the EPSG:4326 axis order - lat/lon.
I was wondering if I use Shapely do I "need" to swap the axis order when creating Shapely objects, and if so - why.
So the question is in what way does Shapely expect the GIS axis order (lon/lat) and not the EPSG:4326 axis order (lat/lon) ?

I was thinking maybe it has to do with the `wkt`, `GeoJSON` input/outputs:
* wkt standard only states x, y and does not say that x=lon and y=lat (though that's usually implied in most GIS software, that use the Plate Carrée projection by default) related discussion: https://github.com/dbeaver/dbeaver/issues/5832#issuecomment-488462622
* GeoJSON - https://geojson.org/geojson-spec.html#positions does state `longitude, latitude, altitude for coordinates in a geographic coordinate reference system`

So by saying that Shapely expects lon/lat order do we mean that we want to be automatically compatible with GeoJSON and that mostly wkt users expect lon/lat axis order, or is there anything else behind that statement ?

**Comment by martinfleis** [comment]
Technically, there's no reason to impose lon/lat to x/y and you are free to treat coordinates the way you wish. However, there are cases when purely geometrical nature of shapely meets outside world. In those cases, it is useful to have an assumption in place allowing us to treat x/y as coordinates with some meaning beyond geometric. 

So you don't _need_ to swap the axis order if you are in a full control of the whole process and know what external libraries like pyproj expect but it may be more comfortable. When you assign lon=x, lat=y, you don't have to think about it anymore (mostly) within the ecosystem as other packages expect it. Also keep in mind that some functions (e.g. plot) would invert your geometries if you don't adhere to this recommendation.

**Comment by idanmiara** [comment]
> In those cases, it is useful to have an assumption in place allowing us to treat x/y as coordinates with some meaning beyond geometric.

Yes, but you have to know the CRS for that - in this case EPSG:4326 (lat/lon) or CRS84 (lon/lat).
In many cases you'd have to project the geometry anyway in order to have some real world meaning like distance or area.
So the question is if Shapely "prefers" in some way CRS84 over EPSG:4326 (for instance, I think that the translation from/to GeoJSON is an example for such a preference, as GeoJSON is always lon/lat). 

> Also keep in mind that some functions (e.g. plot) would invert your geometries if you don't adhere to this recommendation.

Yes, that's another good point - Even though that in order to plot a geographic shape (lat/lon or lon/lat) on a planar graph (x/y) you have to project it - most GIS software by default use the Plate Carrée projection (lon=x, lat=y) so that's another reason to "not have to worry about it" if follow the recommendation.

**Comment by martinfleis** [comment]
Well, you are right but, strictly speaking, shapely doesn't know anything about projections, so the issue with EPSG:4326 (lat/lon) or CRS84 (lon/lat) is technically out of scope (with the exception of GeoJSON I/O).

> So the question is if Shapely "prefers" in some way CRS84 over EPSG:4326

Apart from the GeoJSON, `__geo_interface__` and similar representations that are meant to be interoperable with packages coming from outside world, shapely doesn't care. When it comes it I/O, the recommendation shall be followed to ensure others can read the data in the expected form.
